src/lib/operators/table_scan.cpp

```diff
@@ -234,6 +234,38 @@ std::unique_ptr<AbstractTableScanImpl> TableScan::create_impl() const {
       return std::make_unique<ColumnVsColumnTableScanImpl>(input_table_left(), left_column_expression->column_id,
                                                            predicate_condition, right_column_expression->column_id);
     }
+
+    // Predicate pattern: <column> <binary predicate_condition> <uncorrelated subquery> (and vice versa)
+    {
+      const auto left_subquery = std::dynamic_pointer_cast<PQPSelectExpression>(left_operand);
+      const auto right_subquery = std::dynamic_pointer_cast<PQPSelectExpression>(right_operand);
+
+      std::shared_ptr<PQPColumnExpression> column_expression;
+      std::shared_ptr<PQPSelectExpression> uncorrelated_subquery;
+      bool flip = false;
+
+      if (left_column_expression && right_subquery && !right_subquery->is_correlated()) {
+        column_expression = left_column_expression;
+        uncorrelated_subquery = right_subquery;
+      } else if (right_column_expression && left_subquery && !left_subquery->is_correlated()) {
+        column_expression = right_column_expression;
+        uncorrelated_subquery = left_subquery;
+        flip = true;
+      }
+
+      if (uncorrelated_subquery) {
+        const auto uncorrelated_select_results =
+            ExpressionEvaluator::populate_uncorrelated_select_results_cache({right_subquery});
```

Comment:
This function should not be used here in my opinion. The entry point into the ExpressionEvaluator should be `evaluate_expression_to_result()`.

`populate_uncorrelated_select_results_cache()` is still a hack, imo, and we should not spread its usage over the code base.

Reply:
How would I get hold of `template <typename Result>` though?